ArcGISRuntimeSDKQt_CppSamples/Scenes/CreateTerrainSurfaceFromLocalTile/CreateTerrainSurfaceFromLocalTile.cpp

```diff
@@ -0,0 +1,133 @@
+// [WriteFile Name=CreateTerrainSurfaceFromLocalTile, Category=Scenes]
+// [Legal]
+// Copyright 2019 Esri.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// [Legal]
+
+#include "CreateTerrainSurfaceFromLocalTile.h"
+
+#include "ArcGISTiledElevationSource.h"
+#include "Scene.h"
+#include "SceneQuickView.h"
+
+#include <QDir>
+#include <QUrl>
+
+#ifdef Q_OS_IOS
+#include <QStandardPaths>
+#endif
+
+using namespace Esri::ArcGISRuntime;
+
+// helper method to get cross platform data path
+namespace {
+  QString defaultDataPath()
+  {
+    QString dataPath;
+
+    #ifdef Q_OS_ANDROID
+      dataPath = "/sdcard";
+    #elif defined Q_OS_IOS
+      dataPath = QStandardPaths::writableLocation(QStandardPaths::DocumentsLocation);
+    #else
+      dataPath = QDir::homePath();
+    #endif
+
+    return dataPath;
+  }
+}
+
+using namespace Esri::ArcGISRuntime;
+
+CreateTerrainSurfaceFromLocalTile::CreateTerrainSurfaceFromLocalTile(QObject* parent /* = nullptr */):
+  QObject(parent),
+  m_scene(new Scene(Basemap::imagery(this), this))
+{
+  // create the MontereyElevation data path
+  // data is downloaded automatically by the sample viewer app. Instructions to download
+  // separately are specified in the readme.
+  const QString montereyTileElevationPath = QString{defaultDataPath() + "/ArcGIS/Runtime/Data/tpk/MontereyElevation.tpk"};
+
+  //Before attempting to add any layers, check that the file for the elevation source exists at all.
+  const bool srcElevationFileExists = QFileInfo::exists(montereyTileElevationPath);
+
+  if(srcElevationFileExists)
+  {
+    //Create the elevation source from the local raster(s). RasterElevationSource can take multiple files as inputs, but in this case only takes one.
+    ArcGISTiledElevationSource* elevationSrc = new ArcGISTiledElevationSource{montereyTileElevationPath, this};
+
+    //When the elevation source is finished loading, call the elevationSrcFinishedLoading callback, so we can tell if it loaded succesfully.
+    connect(elevationSrc, &ArcGISTiledElevationSource::doneLoading, this, &CreateTerrainSurfaceFromLocalTile::elevationSrcFinishedLoading, Qt::UniqueConnection);
+
+    // add the elevation source to the scene to display elevation
+    m_scene->baseSurface()->elevationSources()->append(elevationSrc);
+  }
+  else
+  {
+    qWarning() << "Could not find file at : " << montereyTileElevationPath << ". Elevation source not set.";
```

Comment:
generally we output everything to `qDebug()` in samples. Not necessarily a porblem to do this but wouldn't be the common pattern